tomb/src/pipelines/banyan_api/auth.rs

```diff
@@ -54,3 +74,114 @@ pub async fn pipeline(command: AuthSubCommand) -> Result<String> {
+    println!("fingerprint bytes: {:?}", fingerprint);
+    let fingerprint = pretty_fingerprint(fingerprint.as_slice());
+
+    // Bytes of the public device key
+    let public_device_key_bytes = public_device_key
+        .export()
+        .await
+        .map_err(ClientError::crypto_error)?;
+
+    // Public device key in PEM format
+    let public_device_key =
+        String::from_utf8(public_device_key_bytes).expect("cant convert key bytes to string");
+
+    // Strip the public key of its new lines
+    let mut stripped_public_key = public_device_key.replace('\n', "");
+    // Strip the public key of its prefix and suffix
+    stripped_public_key = stripped_public_key
+        .strip_prefix("-----BEGIN PUBLIC KEY-----")
+        .expect("unable to strip PEM prefix")
+        .strip_suffix("-----END PUBLIC KEY-----")
+        .expect("unable to strip PEM suffix")
+        .to_string();
+
+    // Represent the weird b64 characters with ones that are url-valid
+    let encoded_public_key = stripped_public_key
+        .replace('+', "-")
+        .replace('/', "_")
+        .replace('=', ".")
+        .to_string();
+
+    // Create a new request object with the nonce
+    let start_regwait = StartRegwait {
+        fingerprint: fingerprint.clone(),
+    };
+    // Create a clone of the client to move into the handle
+    let mut client_1 = client.clone();
+    // Create a join handle for later use, starting the call immediately
+    let join_handle: JoinHandle<Result<StartRegwaitResponse, ClientError>> =
+        tokio::spawn(async move {
+            // Build the request
+            client_1.call(start_regwait).await
+        });
+
+    // Give the core server a bit of time to start the call
+    std::thread::sleep(Duration::from_secs(3));
+
+    // Should be this in prod TODO
+    // https://alpha.data.banyan.computer/
+
+    let global = GlobalConfig::from_disk().await?;
+
+    // Open this url with firefox
+    open::with(
+        format!(
+            "{}/api/auth/device/register?spki={}",
+            global.remote_frontend, encoded_public_key
+        ),
+        "firefox",
+    )
+    .expect("failed to open browser");
+
+    // Give the server a bit of time to update the db
+    std::thread::sleep(Duration::from_secs(3));
+
+    // Ask the server to pop the fingerprint from the AppState and check that it now exists in SQL db while doing so
+    match client.call_no_content(EndRegwait { fingerprint }).await {
+        // If the end call succeeded
+        Ok(_) => {
+            //
+            let start_response = join_handle
+                .await
+                .map_err(anyhow::Error::new)
+                .map(|v| v.map_err(anyhow::Error::new))??;
+
+            // Update the client's credentials
+            Ok(Credentials {
+                account_id: start_response.account_id,
+                signing_key: private_device_key,
+            })
+        }
+        Err(_) => Err(anyhow::anyhow!("end_regwait failed")),
+    }
```

Comment:
This shouldn't be making a call to the endregwait, that should be getting made by the frontend or tomb-wasm after the key has been registered in the database. This just needs to wait for the start call to complete (which will give you the account ID) or for it to timeout/be rejected (which will return an error).